packages/cli/src/events/event-relay.ts

```diff
@@ -0,0 +1,22 @@
+import { EventService } from './event.service';
+import { Service } from 'typedi';
+import type { RelayEventMap } from '@/events/relay-event-map';
+
+@Service()
+export class EventRelay {
+	constructor(readonly eventService: EventService) {}
+
+	protected setupListeners<EventNames extends keyof RelayEventMap>(map: {
+		[EventName in EventNames]?: (event: RelayEventMap[EventName]) => void | Promise<void>;
+	}) {
+		for (const [eventName, handler] of Object.entries(map) as Array<
+			[EventNames, ((event: RelayEventMap[EventNames]) => void | Promise<void>) | undefined]
+		>) {
+			if (!handler) continue;
```

Comment:
Is there a use case for leaving the `handler` as undefined? I'd imagine it's a bug, so maybe it's better to explode here, e.g. by using an `assert`

Reply:
So I added `undefined` in case someone forgets to specify a handler, but on second thought the type system will catch this oversight beforehand at call site, so we can simplify the type and remove the check.